src/main/scala/scala/collection/decorators/IteratorDecorator.scala

```diff
@@ -72,4 +72,44 @@ class IteratorDecorator[A](val `this`: Iterator[A]) extends AnyVal {
     loop(immutable.List.empty)
   }
 
+  /**
+    * Constructs an iterator where consecutive elements are accumulated as
+    * long as the output of f for each element doesn't change.
+    * <pre>
+    * Vector(1,2,2,3,3,3,2,2)
+    *     .iterator
+    *     .groupUntilChanged(identity)
+    *     .toList
+    * </pre>
+    * produces
+    * <pre>
+    * List(Seq(1),
+    *      Seq(2,2),
+    *      Seq(3,3,3),
+    *      Seq(2,2))
+    * </pre>
+    * @param f the function to compute a key for an element
+    * @tparam K the type of the computed key
+    * @return an iterator of sequences of the consecutive elements with the
+    *         same key in the original iterator
+    */
+  def groupUntilChanged[K](f: A => K): Iterator[Iterable[A]] =
+    new AbstractIterator[Seq[A]] {
+      private var hd: Option[A] = `this`.nextOption()
+      override def hasNext: Boolean = hd.isDefined
+
+      override def next(): Seq[A] = {
+        hd match {
+          case None => Iterator.empty.next()
+          case Some(head) =>
+            hd = `this`.nextOption()
```

Comment:
Most traversals in the standard library use the `hasNext`/`next` calls to avoid allocating `Option` objects, we could do the same thing here :-). Then we can also avoid the function allocation inside the `exists` call on line `107` 😄

Reply:
I understand however that would force me to use null for hd when hasNext is `false` is that acceptable ?

Reply:
No, this won't work because `null` can be a valid element of `A`. I'm also just realizing now that evaluating ``` `this`.nextOption() ``` at all is probably not kosher though, since it violates the laziness of `Iterator`s. I reckon the first call to ``` `this`.next() ```shouldn't happen until the first call tho `next()`.

Reply:
when I talk about using `null`, I am thinking of somethin akin to   https://github.com/scala/scala/blob/2.13.x/src/library/scala/collection/Iterator.scala#L122

Reply:
Ah yes, that'll work!

Reply:
I have tried to address your point, I will push a fixup commit with my attempt at this... I have tried to keep it as simple as possible, it passes the tests but I find it much much less satisfying than the original implementation.